File: app/src/main/java/org/stepic/droid/util/resolvers/StepTypeResolver.java
package org.stepic.droid.util.resolvers;

import android.graphics.drawable.Drawable;

import org.jetbrains.annotations.NotNull;
import org.stepic.droid.base.StepBaseFragment;
import org.stepic.droid.model.Step;

public interface StepTypeResolver {

    Drawable getDrawableForType(String type, boolean viewed, boolean isPeerReview);

    @NotNull
    StepBaseFragment getFragment(Step step);
}

File: app/src/main/java/org/stepic/droid/util/resolvers/StepTypeResolver.kt
package org.stepic.droid.util.resolvers

import org.stepic.droid.ui.quiz.QuizDelegate
import org.stepik.android.model.Step

interface StepTypeResolver {

    fun getDrawableForType(type: String?, isPeerReview: Boolean): Pair<Int, Int>

    fun getQuizDelegate(step: Step?): QuizDelegate
}
